src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -2722,9 +2725,8 @@ const bool java_bytecode_convert_methodt::is_method_inherited(
         return true;
       // methods with the default access modifier are only
       // accessible within the same package.
-      else if(access==ID_default &&
-              java_class_to_package(id2string(parent))==classpackage)
-        return true;
+      else if(access==ID_default)
+        return java_class_to_package(id2string(parent))==classpackage;
```

Comment:
I think the original code was correct, and so, this change introduces a bug. Here are the reasons:

- The loop is iterating through all the ancestor classes in the inheritance tree of class `classname`.
- The method should return `true` iff there exists one class in such tree where the method is accessible in that class.
- When we find a class in a different package with a `default` modifier access, that is equivalent to being private, and the loop should proceed the same way as in the case where it finds a `private` modifier: ignore and continue the search. This is what the original code does.